### PR TITLE
Personal Info with new Design, hmis profiles and living situation (DEV-166)

### PR DESCRIPTION
**Pull request by Davit-BetterAngels**
DEV-166

left a comment on the ticket to create another ticket for profile photo as currently it is not available

### CODE REVIEW

**Comment by shiozakiyuko** [comment]
Hi @Davit-BetterAngels 

- When the Full Name section gets reset and be empty on the all 4 fields and the user tended to save it, currently I don't see the validation message. Can you make the it visible as it shows in [the design](https://www.figma.com/design/waIeX8dknHKedTrVxCQAAG/HIFI---1st-MVPs?node-id=15564-91810&t=0U9KSgcJFDQ8LW6R-4)? "Filling out one of the fields is required"
- When I pressed Update button after made some changes, I thought nothing happened until I figured I had an empty section on top of the page (Full Name section was empty). Haven't we fixed this type of situation before? When the user has an error somewhere in the page, and they tapped Submit button, it should bring them to the area where it has a validation message? 
- I believe we do need another ticket to clean them up... We don't need Gender/HMIS/ Contact Info/ Veteran Status since they belong to different sections now (minus Veteran Status - we omitted that now). But FYI a ticket is coming on your way.
- The calculation of the age is not correct. 1980 should be 44 years old, not 24 (I want the calculation on my age! haha)

Let me know if you have some comments or questions. Thanks!

image
image
image

**Comment by Davit-BetterAngels** [comment]
> Hi @Davit-BetterAngels
> 
> * When the Full Name section gets reset and be empty on the all 4 fields and the user tended to save it, currently I don't see the validation message. Can you make the it visible as it shows in [the design](https://www.figma.com/design/waIeX8dknHKedTrVxCQAAG/HIFI---1st-MVPs?node-id=15564-91810&t=0U9KSgcJFDQ8LW6R-4)? "Filling out one of the fields is required"
> * When I pressed Update button after made some changes, I thought nothing happened until I figured I had an empty section on top of the page (Full Name section was empty). Haven't we fixed this type of situation before? When the user has an error somewhere in the page, and they tapped Submit button, it should bring them to the area where it has a validation message?
> * We don't need Gender/HMIS/ Contact Info/ Veteran Status since they belong to different sections now (minus Veteran Status - we omitted that now). So I believe we do need another ticket to clean them up... I will discuss with @kimchoi-jjiggae and create a ticket for it.
> * The calculation of the age is not correct. 1980 should be 44 years old, not 24 (I want the calculation on my age! haha)
> 
> Let me know if you have some comments or questions. Thanks!
> 
> image image image

1. no field validation in AC
2. we don't have that kind of logic
3. not part of the ticket to remove old ones
4. will take a look :)

**Comment by Davit-BetterAngels** [comment]
@shiozakiyuko age fixed

**Comment by shiozakiyuko** [comment]
Hey @Davit-BetterAngels 

I confirmed the age is fixed now. Thanks! 

As for 1, I added the validation message in the AC.
As for 2, ok gotcha. I may be thinking about somewhere else like my old work... As long as we have the validation message set up, I will skip this one for now.
As for 3, I created a ticket to remove them from the current page but I asked @kimchoi-jjiggae to review the ticket for the moment. 

Thanks so much!

**Comment by mikefeldberg** [comment]
@Davit-BetterAngels `profilePhoto` is available on `ClientProfileType`

**Comment by shiozakiyuko** [comment]
Hey @Davit-BetterAngels 

I just had to test one more time on the Relevant Contact section for **Reset/Remove functionalities**.

Currently, the Reset and Remove buttons act the same (deleting the entire Father card, or Aunt card, for example.
**Reset** should just clear the each input field and reset it as initial state. For example, if it's text field, it should be empty with the placeholder text visible.
**Remove** should literally remove the whole card. For example, if you create Aunt contact info, and tap the remove button, the whole entire card should be deleted. 

I thought the Reset and Remove were working as expected earlier... but that's what I'm experiencing right now.

Please fix. :) Thanks!

**Comment by mikefeldberg** [comment]
> Hey @Davit-BetterAngels
> 
> I confirmed the age is fixed now. Thanks!
> 
> As for 1, I added the validation message in the AC. As for 2, ok gotcha. I may be thinking about somewhere else like my old work... As long as we have the validation message set up, I will skip this one for now. As for 3, I created a ticket to remove them from the current page but I asked @kimchoi-jjiggae to review the ticket for the moment.
> 
> Thanks so much!

@shiozakiyuko validation was added to AC this morning, when this ticket was already in review

**Comment by Davit-BetterAngels** [comment]
> Hey @Davit-BetterAngels
> 
> I just had to test one more time on the Relevant Contact section for **Reset/Remove functionalities**.
> 
> Currently, the Reset and Remove buttons act the same (deleting the entire Father card, or Aunt card, for example. **Reset** should just clear the each input field and reset it as initial state. For example, if it's text field, it should be empty with the placeholder text visible. **Remove** should literally remove the whole card. For example, if you create Aunt contact info, and tap the remove button, the whole entire card should be deleted.
> 
> I thought the Reset and Remove were working as expected earlier... but that's what I'm experiencing right now.
> 
> Please fix. :) Thanks!

for remove and reset. 

they do not do the same. 
one is removing entire section the other one resetting all the fields and you should select relationship again.

because the relationship is a field by itself. If we want another flow we can do it

**Comment by Davit-BetterAngels** [comment]
> @Davit-BetterAngels `profilePhoto` is available on `ClientProfileType`

I know. it is just now available in this PR.

File: libs/expo/betterangels/src/lib/screens/AddEditClient/AddEditClient.graphql [hunk 1]
```diff
@@ -18,15 +18,22 @@ query GetClientProfile($id: ID!) {
   clientProfile(pk: $id) {
     ... on ClientProfileType {
       id
+      age
       address
       dateOfBirth
       gender
       hmisId
```
Comment (mikefeldberg) [inline libs/expo/betterangels/src/lib/screens/AddEditClient/AddEditClient.graphql:25]: please remove `hmisId` from all queries and mutations. `hmisId` should only be accessed via `hmisProfiles`, and not directly on/from clientprofile

File: libs/expo/betterangels/src/lib/screens/AddEditClient/index.tsx [hunk 2]
```diff
@@ -186,12 +184,22 @@ export default function AddEditClient({ id }: { id?: string }) {
...
+
     const newInput = clientInput.contacts?.map((contact) => {
       const { __typename, ...rest } = contact;
       return rest;
     });
 
-    methods.reset({ ...clientInput, contacts: newInput });
+    methods.reset({
+      ...clientInput,
+      contacts: newInput,
```
Comment (mikefeldberg) [inline libs/expo/betterangels/src/lib/screens/AddEditClient/index.tsx:200]:
```suggestion
      contacts: newContacts,
```

File: libs/expo/betterangels/src/lib/screens/Client/Profile/index.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { Colors } from '@monorepo/expo/shared/static';
+import { useState } from 'react';
+import { MainScrollContainer } from '../../../ui-components';
+import { ClientProfileQuery } from '../__generated__/Client.generated';
+import PersonalInfo from './PersonalInfo';
+
+export default function Profile({
+  client,
+}: {
+  client: ClientProfileQuery | undefined;
```
Comment (mikefeldberg) [inline libs/expo/betterangels/src/lib/screens/Client/Profile/index.tsx:10]: Can you please explain what's happening here? Is `client` the `ClientProfileQuery`? Or is it the response of the `ClientProfileQuery`? I'm just not familiar with this syntax 🙏

**Review by mikefeldberg — CHANGES_REQUESTED** [review]
looking good! a few small comments

File: libs/expo/betterangels/src/lib/screens/Client/Profile/index.tsx (hunk 3 above)
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/screens/Client/Profile/index.tsx:10]: either we should pass data.clientProfile and type es ClientProfileType or if we want to pass full response we give ClientProfileQuery Type.

File: libs/expo/betterangels/src/lib/screens/AddEditClient/index.tsx (hunk 2 above)
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/screens/AddEditClient/index.tsx:200]: done

File: libs/expo/betterangels/src/lib/screens/AddEditClient/AddEditClient.graphql (hunk 1 above)
Comment (Davit-BetterAngels) [reply libs/expo/betterangels/src/lib/screens/AddEditClient/AddEditClient.graphql:25]: done

File: libs/expo/betterangels/src/lib/screens/Client/Profile/index.tsx (hunk 3 above)
Comment (mikefeldberg) [reply libs/expo/betterangels/src/lib/screens/Client/Profile/index.tsx:10]: so, when it's used `PersonalInfo`, this is using the ClientProfileQuery Type?
```
  const fullName = `${client?.clientProfile.user.firstName ?? ''} ${
    client?.clientProfile.user.lastName ?? ''
  }`.trim();
```

**Review by mikefeldberg — APPROVED** [review]
🎉

**Pull request merged**